packages/forms-web-app/src/pages/examination/_middleware/examination.middleware.js

```diff
@@ -0,0 +1,29 @@
+const { buildQueryString } = require('../../_utils/build-query-string');
+const { hasExaminationSession } = require('../_session/examination-session');
+const {
+	routesConfig: {
+		examination: {
+			baseDirectory,
+			pages: { haveYourSay }
+		}
+	}
+} = require('../../../routes/config');
+const { getProjectsURL } = require('../../projects/_utils/get-projects-url');
+
+const examinationMiddleware = (req, res, next) => {
+	const { params, path, query, session } = req;
+	const { case_ref } = params;
+
+	if (path.replace('/', '') === haveYourSay.route) return next();
+	else if (!hasExaminationSession(session)) {
+		const projectsURL = getProjectsURL(case_ref);
+		const url = `${projectsURL}/${baseDirectory}/${haveYourSay.route}`;
+		const queryString = Object.keys(query).length ? buildQueryString(query) : '';
```

Comment:
Presumably the query string is to maintain ?lang=cy.  Might be nice to put that in the test case.